### ISSUE TITLE
Emit attribute and createEmpty not working

### ISSUE THREAD

**Issue by MangelMaxime** [fable-compiler/Fable#792]
### Description

Using Emit attribute on type member and using it with `createEmpty` output an empty js object.

### FSharp code

```fs
type TextStyle =
        [<Emit("fontFamilly")>] abstract FontFamilly : string option with get, set
        [<Emit("fontSize")>] abstract FontSize : float option with get, set

let TextStyleDefault =
      let style = createEmpty<Fable.Import.PIXI.TextStyle>
      style.FontFamilly <- Some "Arial"
      style.FontSize <- Some 14.
      style
```

### Actual results

```fs
var TextStyleDefault = __exports.TextStyleDefault = function () {
      var style = {};
      return style;
    }();
```

### Expected results

```fs
var TextStyleDefault = __exports.TextStyleDefault = function () {
      var style = {};
      style.fontFamilly = "Arial";
      style.fontSize = 14;
      return style;
    }();
```

**Comment by alfonsogarciacaro** [comment]
Sorry, it's not very obvious but with attributes you should use the optional parameter syntax. Setters have three parameters: this, property name and value, while getters only have two:

```fsharp
type TextStyle =
        [<Emit("$0.fontFamily{{=$2}}")>]
        abstract FontFamilly : string option with get, set
        [<Emit("$0.fontSize{{=$2}}")>]
        abstract FontSize : float option with get, set

let TextStyleDefault =
      let style = createEmpty<Fable.Import.PIXI.TextStyle>
      style.FontFamilly <- Some "Arial"
      style.FontSize <- Some 14.
      style
```
Reference: http://fable.io/docs/interacting.html#Emit-attribute

**Comment by alfonsogarciacaro** [comment]
Hmm, actually I just tried with Fable 1.0 and it's not working :( I need to fix it.

**Comment by MangelMaxime** [comment]
Ah sorry didn't see this para in the doc :) thanks for pointed it out.

And ok will wait your fix so :)

**Comment by Zaid-Ajaj** [comment]
Without `Emit` attributes it works :)

```fsharp
open Fable.Core
open Fable.Core.JsInterop

type TextStyle =
    abstract FontFamilly : string option with get, set
    abstract FontSize : float option with get, set

let TextStyleDefault =
      let style = createEmpty<TextStyle>
      style.FontFamilly <- Some "Arial"
      style.FontSize <- Some 14.
      style
```

See [Sample here](http://fable.io/repl#?code=open%20Fable.Core%0Aopen%20Fable.Core.JsInterop%0A%0Atype%20TextStyle%20%3D%0A%20%20%20%20abstract%20FontFamilly%20%3A%20string%20option%20with%20get%2C%20set%0A%20%20%20%20abstract%20FontSize%20%3A%20float%20option%20with%20get%2C%20set%0A%0Alet%20TextStyleDefault%20%3D%0A%20%20%20%20%20%20let%20style%20%3D%20createEmpty%3CTextStyle%3E%0A%20%20%20%20%20%20style.FontFamilly%20%3C-%20Some%20%22Arial%22%0A%20%20%20%20%20%20style.FontSize%20%3C-%20Some%2014.%0A%20%20%20%20%20%20style)

**Comment by MangelMaxime** [comment]
@Zaid-Ajaj Yes but we have the `FontFamily` and not `fontFamily`.
Note the first letter as a lower one.

**Comment by Zaid-Ajaj** [comment]
@MangelMaxime What about [this one](http://fable.io/repl#?code=open%20Fable.Core%0Aopen%20Fable.Core.JsInterop%0A%0Atype%20TextStyle%20%3D%0A%20%20%20%20%5B%3CEmit(%22%240.fontFamily%20%3D%20%241%22)%3E%5D%0A%20%20%20%20abstract%20FontFamilly%20%3A%20string%20option%20with%20get%2C%20set%0A%20%20%20%20%5B%3CEmit(%22%240.fontSize%20%3D%20%241%22)%3E%5D%0A%20%20%20%20abstract%20FontSize%20%3A%20float%20option%20with%20get%2C%20set%0A%0Alet%20TextStyleDefault%20%3D%0A%20%20%20%20%20%20let%20style%20%3D%20createEmpty%3CTextStyle%3E%0A%20%20%20%20%20%20style.FontFamilly%20%3C-%20Some%20%22Arial%22%0A%20%20%20%20%20%20style.FontSize%20%3C-%20Some%2014.%0A%20%20%20%20%20%20style)?

using `[<Emit("$0.propName = $1")>]` seems to work well

**Comment by MangelMaxime** [comment]
@Zaid-Ajaj Seems to work for me too :)

Always the savior of `Emit` usage :D

@alfonsogarciacaro I let you decide if the proposition of @Zaid-Ajaj is enough to fix this issue or not :)

**Comment by Zaid-Ajaj** [comment]
@MangelMaxime I just realized it only works for `get`ing the property :(

**Comment by Zaid-Ajaj** [comment]
OK the final solution with optional parameters works for both `set` and `get` :smile:
```fsharp 
open Fable.Core
open Fable.Core.JsInterop

type TextStyle =
    [<Emit("{{$1? $0.fontFamilly = $1 : $0.fontFamilly}}")>]
    abstract FontFamilly : string option with get, set
    
    [<Emit("{{$1? $0.fontSize = $1 : $0.fontSize}}")>]
    abstract FontSize : float option with get, set

let TextStyleDefault =
      let style = createEmpty<TextStyle>
      style.FontFamilly <- Some "Arial"
      style.FontSize <- Some 14.
      let family = style.FontFamilly
      let size = style.FontSize
      style
```

Compiles to 

```javascript
export const TextStyleDefault = (() => {
  const style = {};
  style.fontFamilly = "Arial";
  style.fontSize = 14;
  const family = style.fontFamilly;
  const size = style.fontSize;
  return style;
})();
```

**Comment by alfonsogarciacaro** [comment]
@Zaid-Ajaj is right, the setter only has two parameters, I mistook it with indexed properties, sorry!

However, as @Zaid-Ajaj also noted, you still need the conditional parameters to make both getter and setter work. But there was an error in `fable-loader` I just fixed, please update to 1.0.0-narumi-906 and use the code in my first reply (I just edited it).

**Comment by alfonsogarciacaro** [comment]
@Zaid-Ajaj I think you're the first person that used the `{{$1?i:}}` syntax ;)

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro I can't get 1.0.0-narumi-906.

I did: 
```
➜  showcase git:(feature/next) ✗ dotnet clean          
Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.

➜  showcase git:(feature/next) ✗ rm -rf bin obj
➜  showcase git:(feature/next) ✗ dotnet fable --version
No executable found matching command "dotnet-fable"
➜  showcase git:(feature/next) ✗ dotnet restore --no-cache
  Restoring packages for /home/maxime/Workspace/Hink/showcase/Showcase.fsproj...
  Restoring packages for /home/maxime/Workspace/Hink/showcase/Showcase.fsproj...
  Restoring packages for /home/maxime/Workspace/Hink/src/Hink.fsproj...
  Restore completed in 1.07 sec for /home/maxime/Workspace/Hink/showcase/Showcase.fsproj.
  Lock file has not changed. Skipping lock file write. Path: /home/maxime/Workspace/Hink/src/obj/project.assets.json
  Restore completed in 3.97 sec for /home/maxime/Workspace/Hink/src/Hink.fsproj.
  Generating MSBuild file /home/maxime/Workspace/Hink/showcase/obj/Showcase.fsproj.nuget.g.props.
  Generating MSBuild file /home/maxime/Workspace/Hink/showcase/obj/Showcase.fsproj.nuget.g.targets.
  Writing lock file to disk. Path: /home/maxime/Workspace/Hink/showcase/obj/project.assets.json
  Restore completed in 4.15 sec for /home/maxime/Workspace/Hink/showcase/Showcase.fsproj.
  
  NuGet Config files used:
      /home/maxime/.nuget/NuGet/NuGet.Config
  
  Feeds used:
      https://api.nuget.org/v3/index.json
➜  showcase git:(feature/next) ✗ dotnet fable --version   
1.0.0-narumi-907
```

But I was already using 1.0.0-narumi-907

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro Have you published the new version of `fable-loader` ?
I can only get version 1.0.0-narumi-905.

**Comment by alfonsogarciacaro** [comment]
@MangelMaxime Yes, I did. But it seems npm is taking a bit to index it... it's Saturday night ;) Can you please try again in 5 mins?

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro Yes sure :)
Sorry ^^

**Comment by Zaid-Ajaj** [comment]
@alfonsogarciacaro they are quite handy 😄

**Comment by alfonsogarciacaro** [comment]
@MangelMaxime @Zaid-Ajaj Sorry, I just found out that Emit is seriously broken :/ 1.0.0-narumi-906 won't fix it. Working on that...

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro Ok no problem, I will rename the member for the moment.
Take your time to fix it :)

**Comment by alfonsogarciacaro** [comment]
@MangelMaxime @Zaid-Ajaj Ok, I've published fable-loader 1.0.0-narumi-907, it should work now 🙏

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro 
Thanks for the fix it's working :)
